addons/google_account/models/google_service.py

```diff
@@ -120,6 +120,41 @@ def _get_google_token_json(self, authorize_code, service):
             error_msg = _("Something went wrong during your token generation. Maybe your Authorization Code is invalid")
             raise self.env['res.config.settings'].get_config_warning(error_msg)
 
+    @api.model
+    def _get_access_token(self, refresh_token, service, scope):
+        """Fetch the access token thanks to the refresh token."""
+        get_config_warning = self.env['res.config.settings'].get_config_warning
+        get_param = self.env['ir.config_parameter'].sudo().get_param
+        client_id = get_param('google_%s_client_id' % service, default=False)
+        client_secret = get_param('google_%s_client_secret' % service, default=False)
+
+        if not client_id or not client_secret:
+            raise get_config_warning(_('Google %s is not yet configured.', service.title()))
```

Comment:
Does not seem necessary since you don't use the special formatting that it provides.
Might as well simply raise UserErrors.